.pipelines/daily_addons_enablement_test.yaml

```diff
@@ -0,0 +1,56 @@
+schedules:
+- cron: '0 0 * * *'
+  displayName: Daily midnight test
+  branches:
+    include:
+    - ci_prod
+jobs:
+    - job: test
+      variables:
+        armServiceConnectionName: 'ci-1es-acr-connection'
+        subscription: '9b96ebbd-c57a-42d1-bbe9-b69296e4c7fb'
+      pool:
+        name: Azure-Pipelines-CI-Test-EO
+      steps:
+          - task: AzureCLI@2
+            inputs:
+              azureSubscription: '${{ variables.armServiceConnectionName }}'
+              scriptType: 'bash'
+              scriptLocation: 'inlineScript'
+              inlineScript: |
+                az --version
+                az account show
+                az account set -s ${{ variables.subscription }}
+                
+                # Define the resource group and AKS cluster name
+                resourceGroup="ci-logs-prod-aks"
+                aksClusterName="daily_enable_disable_test"
+                logAnalyticsWorkspace="/subscriptions/9b96ebbd-c57a-42d1-bbe9-b69296e4c7fb/resourcegroups/ci-logs-prod-aks/providers/microsoft.operationalinsights/workspaces/daily-enable-disable-test-la"
+                workspaceId="0b525f9a-755c-4a1a-b6fe-f96a23ea8a4e"
+
+                # Check if the omsagent addon is enabled
+                addonEnabled=$(az aks show --resource-group $resourceGroup --name $aksClusterName --query addonProfiles.omsagent.enabled -o tsv)
+                
+                if [ "$addonEnabled" == "true" ]; then
+                    echo "OMSAgent addon is enabled on the AKS cluster."
+                    az aks disable-addons --addons monitoring --name $aksClusterName --resource-group $resourceGroup
+                    sleep 10
+                    echo "Re-enabling the OMSAgent addon..."
+                    az aks enable-addons --addons monitoring --name $aksClusterName --resource-group $resourceGroup --workspace-resource-id $logAnalyticsWorkspace
+                    echo "OMSAgent addon has been re-enabled."
+
+                    echo "Sleeping 6 minutes for everything to get set up and logs start flowing"
+                    sleep 360

```

Comment:
make these intervals as pipeline variables so that these can tuned without changing the script.